MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,1220 @@ If the requested :code:`timeout_nsec` exceeds
+  method returns with an error return code of
+  :code:`C.YDB_ERR_TIME2LONG`.
+- If it is able to aquire the lock resource within :code:`timeoutNsec`
+  nanoseconds, it returns with a return code of :code:`C.YDB_OK`;
+  otherwise it returns with a return code of
+  :code:`C.YDB_LOCK_TIMEOUT`. If :code:`timeoutNsec` is zero, the
+  method makes exactly one attempt to acquire the lock.
+- Other errors return an appropriate `error return code`_ prefixed
+  with :code:`C.`.
+
+Go Comprehensive API
+====================
+
+The Go Comprehensive API is a project for the future, to follow the C
+`Comprehensive API`_
+
+Go Utility Functions
+====================
+
+---------
+Go Exit()
+---------
+
+.. code-block:: go
+
+    yottadb.Exit() int
+
+For a process that wishes to close YottaDB databases and no longer use
+YottaDB, the function wraps `ydb_exit()`_ so that any further calls to
+YottaDB result in a :code:`C.YDB_ERR_CALLINAFTEREXIT` error.
+
+Typical processes will not need to call :code:`Exit()` because
+normal process termination closes databases cleanly. However, a
+process that has completed its use of YottaDB, but intends to continue
+with other work for some non-trivial period of time, should call
+:code:`Exit()`.
+
+See also `Go ForkExec()`_.
+
+-------------
+Go ForkExec()
+-------------
+
+.. code-block:: go
+
+    yottadb.ForkExec(argv0 string, argv []string, attr *ProcAttr) int, int
+
+The function has the same signature as the `Go
+syscall.ForkExec() <https://golang.org/pkg/syscall/#ForkExec>`_, which
+it wraps. The YottaDB `Go ForkExec()`_ ensures that the child process
+is safely disconnected from YottaDB interprocess communication
+resources and database files.
+
+Application code that must use :code:`syscall.Forkexec()` should call
+`Go Exit()`_ first, which means that after calling
+:code:`syscall.ForkExec()`, a process can no longer call the YottaDB
+runtime system.
+
+--------------
+Go ForkNCore()
+--------------
+
+.. code-block:: go
+
+    yottadb.ForkNCore()
+
+The function wraps `ydb_fork_n_core()`_ to generate a core file
+snapshot of a process for debugging purposes. See `ydb_fork_n_core()`_
+for more information.
+
+:code:`ForkNCore()` has no parameters and returns no return code.
+
+---------
+Go Free()
+---------
+
+.. code-block:: go
+
+    yottadb.Free(ptr unsafe.pointer)
+
+The function wraps `ydb_free()`_ to release memory previously
+allocated using :code:`Malloc()`. As passing a :code:`ptr` not
+previously allocated using :code:`Malloc()` will result in
+unpredictable behavior, application code should be written with an
+appropriate level of diligence when calling :code:`Free()`.
+
+---------
+Go Init()
+---------
+
+.. code-block:: go
+
+    yottadb.Init() int
+
+The function wraps `ydb_init()`_ to initialize the YottaDB runtime
+system. This call is normally not required as YottaDB initializes
+itself on its first call, the exception being when an application
+wishes to set its own signal handlers (see
```

Comment:
I haven't read the Signals chapter in awhile (on my list) but I'm not entire sure the need to call ydb_init() during signal setup is true anymore. The plan is to modify signal setup in YottaDB so that if a handler is defined, it is recorded. Then if that signal occurs, once we are done with YottaDB's processing of the signal, we will call the handler that was previously setup (Golang's handler). If handlers are replaced after YottaDB is initialized, yottadb won't know about it (except for SIGALRM which it specifically checks). I don't know if Golang plays well with others in terms of signals so best would be to have all the handlers setup before YottaDB is initialized. Note this handling is not Golang specific and would apply to an non-M main routine.

Reply:
@nars1, I'll go with whatever you and @estess agree is right for the document.